code/game/objects/items/machine_wand.dm

```diff
@@ -0,0 +1,181 @@
+	. = ..()
+	if(controlling_machine_or_bot)
+		return controlling_machine_or_bot.ui_act(action, params, ui, state)
+
+/obj/item/machine_remote/AltClick(mob/user)
+	. = ..()
+	if(moving_bug) //we have a bug in transit, so let's kill it.
+		QDEL_NULL(moving_bug)
+	if(!controlling_machine_or_bot)
+		return
+	say("Remote control over [controlling_machine_or_bot] stopped.")
+	remove_old_machine()
+
+/obj/item/machine_remote/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, timeout_time))
+		playsound(src, 'sound/machines/synth_no.ogg', 30 , TRUE)
+		say("Remote control disabled temporarily. Please try again soon.")
+		return FALSE
+	if(!ismachinery(target) && !isbot(target))
+		return
+	if(moving_bug) //we have a bug in transit already, so let's kill it.
+		QDEL_NULL(moving_bug)
+	var/turf/spawning_turf = (controlling_machine_or_bot ? get_turf(controlling_machine_or_bot) : get_turf(src))
+	moving_bug = new(spawning_turf, src, target)
+	remove_old_machine()
+
+///Sets a controlled machine to a new machine, if possible. Checks if AIs can even control it.
+/obj/item/machine_remote/proc/set_controlled_machine(obj/machinery/new_machine)
+	if(controlling_machine_or_bot == new_machine)
+		return
+	remove_old_machine()
+	if(istype(new_machine, /obj/machinery/power/apc))
+		var/obj/machinery/power/apc/new_apc = new_machine
+		if(new_apc.aidisabled)
+			say("AI wire cut, machine uncontrollable.")
+			return
+	else if(istype(new_machine, /obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/new_airlock = new_machine
+		if(!new_airlock.canAIControl())
+			say("AI wire cut, machine uncontrollable.")
+			return
+	controlling_machine_or_bot = new_machine
+	controlling_machine_or_bot.add_overlay(bug_appearance)
+	RegisterSignal(controlling_machine_or_bot, COMSIG_QDELETING, PROC_REF(on_control_destroy))
+	RegisterSignal(controlling_machine_or_bot, COMSIG_ATOM_EMP_ACT, PROC_REF(on_machine_emp))
+
+///Removes the machine being controlled as the current machine, taking its signals and overlays with it.
+/obj/item/machine_remote/proc/remove_old_machine()
+	if(!controlling_machine_or_bot)
+		return
+	UnregisterSignal(controlling_machine_or_bot, list(COMSIG_ATOM_EMP_ACT, COMSIG_QDELETING))
+	controlling_machine_or_bot.cut_overlay(bug_appearance)
+	controlling_machine_or_bot = null
+
+///Called when the machine we're controlling is EMP, removing our control from it.
+/obj/item/machine_remote/proc/on_machine_emp(datum/source, severity, protection)
+	SIGNAL_HANDLER
+	if(severity & EMP_PROTECT_CONTENTS)
+		return
+	disable_remote(EMP_TIMEOUT_DURATION)
+
+/obj/item/machine_remote/proc/disable_remote(timeout_duration)
+	remove_old_machine()
+	COOLDOWN_START(src, timeout_time, timeout_duration)
+
+///The effect of the bug moving towards the selected machinery to mess with.
+/obj/effect/bug_moving
+	name = "bug"
+	desc = "Where da bug goin?"
+	icon_state = "fly"
+	obj_flags = CAN_BE_HIT
+	max_integrity = 100
```

Comment:
That's a lot of integrity for something almost as small as a pixel moving around the screen.